### ISSUE TITLE
Exception after upgrade: An IL variable is not available at the current native IP. 

### ISSUE THREAD

**Issue by jaklithn** [dotnet/runtime#92777]
I had a working .NET 8 Perview 7 WebAssembly Blazor application developed in Rider. After installing .NET 8 RC1 I upgraded all my Nuget packages. I read the documentation but found no direct reasons to adapt my code. I upgraded Rider to version 2023.2.2.

When launching my application I get:
`System.ExecutionEngineException: Exception of type 'System.ExecutionEngineException' was thrown.`
`An IL variable is not available at the current native IP. The error code is CORDBG_E_IL_VAR_NOT_AVAILABLE, or 0x80131304.`

In the console I have this stacktrace:
```
Fatal error. Internal CLR error. (0x80131506)
   at System.Runtime.Serialization.SerializationGuard.<ThrowIfDeserializationInProgress>g__ThrowIfDeserializationInProgress|0_0(System.Runtime.Serialization.SerializationInfo, System.String, Int32 ByRef)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(System.Diagnostics.ProcessStartInfo)
   at Microsoft.AspNetCore.Builder.DebugProxyLauncher+<LaunchAndGetUrl>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Builder.DebugProxyLauncher+<LaunchAndGetUrl>d__9, Microsoft.AspNetCore.Components.WebAssembly.Server, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<LaunchAndGetUrl>d__9 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Start[[Microsoft.AspNetCore.Builder.DebugProxyLauncher+<LaunchAndGetUrl>d__9, Microsoft.AspNetCore.Components.WebAssembly.Server, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<LaunchAndGetUrl>d__9 ByRef)
   at Microsoft.AspNetCore.Builder.DebugProxyLauncher.LaunchAndGetUrl(System.IServiceProvider, System.String, Boolean)
   at Microsoft.AspNetCore.Builder.DebugProxyLauncher.EnsureLaunchedAndGetUrl(System.IServiceProvider, System.String, Boolean)
   at Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions+<>c+<<UseWebAssemblyDebugging>b__0_1>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions+<>c+<<UseWebAssemblyDebugging>b__0_1>d, Microsoft.AspNetCore.Components.WebAssembly.Server, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<<UseWebAssemblyDebugging>b__0_1>d ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions+<>c+<<UseWebAssemblyDebugging>b__0_1>d, Microsoft.AspNetCore.Components.WebAssembly.Server, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<<UseWebAssemblyDebugging>b__0_1>d ByRef)
   at Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions+<>c.<UseWebAssemblyDebugging>b__0_1(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware+<InvokeCore>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Builder.Extensions.MapMiddleware+<InvokeCore>d__4, Microsoft.AspNetCore.Http.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<InvokeCore>d__4 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Builder.Extensions.MapMiddleware+<InvokeCore>d__4, Microsoft.AspNetCore.Http.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<InvokeCore>d__4 ByRef)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.InvokeCore(Microsoft.AspNetCore.Http.HttpContext, Microsoft.AspNetCore.Http.PathString, Microsoft.AspNetCore.Http.PathString)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware+<Invoke>d__11.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Authorization.AuthorizationMiddleware+<Invoke>d__11, Microsoft.AspNetCore.Authorization.Policy, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__11 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Authorization.AuthorizationMiddleware+<Invoke>d__11, Microsoft.AspNetCore.Authorization.Policy, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__11 ByRef)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware+<Invoke>d__6.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Authentication.AuthenticationMiddleware+<Invoke>d__6, Microsoft.AspNetCore.Authentication, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__6 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Authentication.AuthenticationMiddleware+<Invoke>d__6, Microsoft.AspNetCore.Authentication, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__6 ByRef)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl+<Invoke>d__14.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl+<Invoke>d__14, Microsoft.AspNetCore.Diagnostics, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__14 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl+<Invoke>d__14, Microsoft.AspNetCore.Diagnostics, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<Invoke>d__14 ByRef)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.HostFiltering.HostFilteringMiddleware.Invoke(Microsoft.AspNetCore.Http.HttpContext)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequests>d__238`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequests>d__238`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], Microsoft.AspNetCore.Server.Kestrel.Core, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<ProcessRequests>d__238`1<System.__Canon> ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequests>d__238`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], Microsoft.AspNetCore.Server.Kestrel.Core, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<ProcessRequests>d__238`1<System.__Canon> ByRef)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Microsoft.AspNetCore.Hosting.Server.IHttpApplication`1<System.__Canon>)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequestsAsync>d__237`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequestsAsync>d__237`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], Microsoft.AspNetCore.Server.Kestrel.Core, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<ProcessRequestsAsync>d__237`1<System.__Canon> ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol+<ProcessRequestsAsync>d__237`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], Microsoft.AspNetCore.Server.Kestrel.Core, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](<ProcessRequestsAsync>d__237`1<System.__Canon> ByRef)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Microsoft.AspNetCore.Hosting.Server.IHttpApplication`1<System.__Canon>)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Stream`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
```

**Comment by mkArtakMSFT** [comment]
@lewing FYI

**Comment by vcsjones** [comment]
Based on the exception this looks like a known bug in RC1. https://github.com/dotnet/runtime/issues/92007

It's fixed in RC2 by https://github.com/dotnet/runtime/pull/90844.

**Comment by jaklithn** [comment]
> @jaklithn Thanks for the report ... is it possible to create a shareable repro (presumably just a tiny part of your app)?

Sorry for late reply. I had so many issues with this so I had to focus on surviving. It took me several hours to get my repo working again. It was not enough to target projects back to NET 8 Preview 7. I also had to completely remove the new SDK from my machine. I am happy to have a working situation. I will hesitate to install NET 8 RCX.

**Comment by SamMonoRT** [comment]
cc @maraf @pavelsavara - please triage and assign milestone accordingly.

**Comment by pavelsavara** [comment]
@thaystg your thoughts ?

**Comment by thaystg** [comment]
This is related to coreclr debugging not wasm debugging. @isadorasophia can you help us with that?

**Comment by vcsjones** [comment]
> Based on the exception this looks like a known bug in RC1. #92007

> It's fixed in RC2 by #90844.

Now that RC2 is out, @jaklithn can you try in RC2 to see if it alleviates the problem?

**Comment by jaklithn** [comment]
I installed VS 17.8.0 Preview 3 to ensure .NET 8 RC2 came in recommended context.
I can confirm it now works fine when launched in VS!
Contrary to what is said in the release notes it actually works in Rider as well.
